### PR TITLE
PWGHF: small fixes in charm baryon tasks

### PR DESCRIPTION
**Pull request by ZFederica**
- change searchUpToQuark to false in xic0/omegac0 candidae creator MC struct
- use dedicated function for MC event selection in toXiPi efficiency task
- switch from pseudorapidity to rapidity in generated variables table (xic0/omegac0 candidate creator)
@fcatalan92

### CODE REVIEW

File: PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx [hunk 1]
```diff
@@ -43,7 +43,8 @@ struct HfTaskMcEfficiencyToXiPi {
   Configurable<int> nClustersTpcMin{"nClustersTpcMin", 70, "Minimum number of TPC clusters requirement for pion <-- charm baryon"};
   Configurable<int> nClustersItsMin{"nClustersItsMin", 3, "Minimum number of ITS clusters requirement for pion <- charm baryon"};
 
-  Configurable<bool> rejGenTFAndITSROFBorders{"rejGenTFAndITSROFBorders", true, "Reject generated particles coming from bc close to TF and ITSROF borders"};
+  Configurable<bool> rejGenTFBorders{"rejGenTFBorders", true, "Reject generated particles coming from bc close to TF borders"};
+  Configurable<bool> rejGenITSROFBorders{"rejGenITSROFBorders", true, "Reject generated particles coming from bc close to ITSROF borders"};
```
Comment (fcatalan92) [inline PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx:47]:
```suggestion
  Configurable<bool> rejGenITSROFBorders{"rejGenITSROFBorders", false, "Reject generated particles coming from bc close to ITSROF borders"};
```

File: PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx [hunk 2]
```diff
@@ -212,13 +213,15 @@ struct HfTaskMcEfficiencyToXiPi {
...
-        }
+      auto coll = mcParticle.template mcCollision_as<aod::McCollisions>();
+      auto bc = coll.template bc_as<BCsInfo>();
+
+      // accept only mc particles coming from bc that are far away from TF border and/or ITSROFrame
+      if (rejGenTFBorders && !bc.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
+        continue;
+      }
+      if (rejGenITSROFBorders && !bc.selection_bit(o2::aod::evsel::kNoITSROFrameBorder)) {
+        continue;
```
Comment (fcatalan92) [inline PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx:224]: You could consider to use the event selection utility recently implemented for MC (also in a future PR)
https://github.com/AliceO2Group/O2Physics/blob/23af5c7dce65a351c40a42f7705fc701961170e3/PWGHF/Utils/utilsEvSelHf.h#L219

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx [hunk 3]
```diff
@@ -758,7 +758,7 @@ struct HfCandidateCreatorXic0Omegac0Mc {
         // Check whether the charm baryon is non-prompt (from a b quark).
         if (flag != 0) {
           auto particle = mcParticles.rawIteratorAt(indexRecCharmBaryon);
-          origin = RecoDecay::getCharmHadronOrigin(mcParticles, particle, true, &idxBhadMothers);
+          origin = RecoDecay::getCharmHadronOrigin(mcParticles, particle, false, &idxBhadMothers);
         }
         if (origin == RecoDecay::OriginType::NonPrompt && idxBhadMothers.size() > 0) {
```
Comment (fcatalan92) [inline PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx:763]:
```suggestion
        if (origin == RecoDecay::OriginType::NonPrompt) {
```
You should be able to drop this part now

**Review by fcatalan92 — CHANGES_REQUESTED** [review]

File: PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx (hunk 2 above)
Comment (ZFederica) [reply PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx]: I implemented it in this PR

File: PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx [hunk 4]
```diff
@@ -34,6 +37,9 @@ using namespace o2::framework;
 using namespace o2::framework::expressions;
 
 struct HfTaskMcEfficiencyToXiPi {
+
+  o2::hf_evsel::HfEventSelectionMc hfEvSelMc; // mc event selection and monitoring
```
Comment (fcatalan92) [inline PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx:41]: I'm not sure if this is enough, you probably need to have also an
```
o2::hf_evsel::HfEventSelection hfEvSel;
```
to inject the configurables in the task

File: PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx [hunk 5]
```diff
@@ -83,6 +87,8 @@ struct HfTaskMcEfficiencyToXiPi {
       LOGF(fatal, "Both processOmegac0 and processXic0 enabled, please choose ONLY one!");
     }
 
+    hfEvSelMc.addHistograms(registry); // particles monitoring
```
Comment (fcatalan92) [inline PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx:90]: You need to call also `configureFromDevice` in the init

**Comment by fcatalan92** [comment]
Hi @ZFederica, thanks for implementing my comments! I think that the MC event selection as it is implemented now is not configurable. I added a couple of comments with more details.

File: PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx [hunk 6]
```diff
@@ -24,8 +24,11 @@
 #include "Common/Core/RecoDecay.h"
 #include "Common/DataModel/TrackSelectionTables.h"
 
+#include "PWGHF/Core/CentralityEstimation.h"
+#include "PWGHF/Core/SelectorCuts.h"
 #include "PWGHF/DataModel/CandidateReconstructionTables.h"
 #include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "PWGHF/Utils/utilsEvSelHf.h"
```
Comment (fcatalan92) [inline PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx:31]: The additional includes could be removed now

**Review by fcatalan92 — CHANGES_REQUESTED** [review]

**Review by fcatalan92 — APPROVED** [review]

**Pull request merged**